databroker/tests/utils.py

```diff
@@ -6,43 +6,24 @@
 
 import tzlocal
 
-from databroker import Broker, BrokerES
+from databroker import Broker, BrokerES, temp_config
 from databroker.headersource import HeaderSourceShim
 from databroker.eventsource import EventSourceShim
 from subprocess import Popen
 
 
 def build_sqlite_backed_broker(request):
     """Uses mongoquery + sqlite -- no pymongo or mongo server anywhere"""
-    from ..headersource.sqlite import MDS
-    from ..assets.sqlite import Registry
-
-    tempdirname = tempfile.mkdtemp()
-    mds = MDS({'directory': tempdirname,
-               'timezone': tzlocal.get_localzone().zone,
-               'version': 1})
-    filenames = ['run_starts.json', 'run_stops.json', 'event_descriptors.json',
-                 'events.json']
-    for fn in filenames:
-        with open(os.path.join(tempdirname, fn), 'w') as f:
-            f.write('[]')
-
-    def delete_mds():
-        shutil.rmtree(tempdirname)
-
-    request.addfinalizer(delete_mds)
 
-    tf = tempfile.NamedTemporaryFile()
-    fs = Registry({'dbpath': tf.name})
+    config = temp_config()
+    tempdir = config['metadatastore']['config']['directory']
 
-    def delete_fs():
-        os.remove(tf.name)
+    def cleanup():
+        os.remove(tempdir)
```

Comment:
It looks like this is trying to remove a (non-empty?) directory and failing.

Reply:
This should be using rmtree anyway. Force-pushed a fix.